src/sidebar/components/menu-keyboard-navigation.js

```diff
@@ -0,0 +1,99 @@
+import { createElement } from 'preact';
+import { useEffect, useRef } from 'preact/hooks';
+import propTypes from 'prop-types';
+
+function isElementVisible(element) {
+  return element.offsetParent !== null;
+}
+
+/**
+ * Menu wrapper component to facilitate keyboard navigation of a list of
+ * <MenuItem> components.
+ *
+ * Note that `ArrowRight` shall be handled by the <MenuItem> directly and
+ * all other focus() related  navigation is handled here.
+ */
+export default function MenuKeyboardNavigation({
+  className,
+  closeMenu,
+  children,
+  isVisible,
+}) {
+  const menuRef = useRef(null);
+
+  useEffect(() => {
+    if (isVisible) {
+      // TODO: figure out something better than this timeout hack if possible.
+      // The focus won't work without delaying rendering.
+      setTimeout(() => {
```

Comment:
Can you explain why this was needed. Effects (both `useEffect` and `useLayoutEffect`) run after the DOM is updated so you should be able to get a reference to the menu item either way.

Reply:
I think this has to do with `display: contentVisible ? '' : 'none',` in slider.js, but I don't have a more appropriate fix at this time